tempesta_fw/t/functional/helpers/control.py

```diff
@@ -236,12 +274,44 @@ def clients_parallel_load(client, count=None):
     client.cleanup()
 
 #-------------------------------------------------------------------------------
-# Tempesta
+# Client
+#-------------------------------------------------------------------------------
+class DeproxyClient(object, Stateful):
+    """ Class for deproxy functional client """
+    def __init__(self, client):
+        self.client = client
+        self.stop_procedures = [self.stop_client]
+
+    def run_start(self):
+        tf_cfg.dbg(3, '\tStarting deproxy client')
+        self.client.tempesta_connect()
+
+    def stop_client(self):
+        tf_cfg.dbg(3, '\tStop deproxy client')
+        asyncore.close_all()
```

Comment:
This is very internal for `deproxy.py`. Also `asyncore.close_all()` stops network activity for  all classes that uses `python.asyncore()`. May be current file structure is outdated with introducing statedull clasess and `control.Tempesta` should be moved to `tempesta.Tempesta`, `control.Nginx` to `nginx.Nginx` and so on? What is your opinion?

Reply:
I'm agree with this

Reply:
but i think, this should be done in separate PR